src/Platforms/AbstractPlatform.php

```diff
@@ -3882,6 +3908,132 @@ public function getTimeTypeDeclarationSQL(array $column)
+    {
+        return $this->getTimeTypeDeclarationSQLSnippet($column)
+             . $this->getTimeTypePrecisionSQLSnippet($column)
+             . $this->getTimeTypeSQLSuffix($column);
+    }
+
+    /**
+     * Obtains DBMS specific SQL declaration snippet to be used to create datetime columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     *
+     * @throws Exception If not supported on this platform.
+     */
+    protected function getDateTimeTypeDeclarationSQLSnippet(array $column): string
+    {
+        throw Exception::notSupported(__METHOD__);
+    }
+
+    /**
+     * Obtains DBMS specific SQL declaration snippet to be used to create datetimetz columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTzTypeDeclarationSQLSnippet(array $column): string
+    {
+        return $this->getDateTimeTypeDeclarationSQLSnippet($column);
+    }
+
+    /**
+     * Obtains DBMS specific SQL declaration snippet to be used to create time columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     *
+     * @throws Exception If not supported on this platform.
+     */
+    protected function getTimeTypeDeclarationSQLSnippet(array $column): string
+    {
+        throw Exception::notSupported(__METHOD__);
+    }
+
+    /**
+     * Obtains DBMS specific suffix to add to SQL declaration to create datetime columns in statements
+     * like CREATE TABLE, for example ' WITHOUT TIME ZONE'.
+     *
+     * Requires a leading space unless empty.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTypeSQLSuffix(array $column): string
+    {
+        return '';
+    }
+
+    /**
+     * Obtains DBMS specific suffix to add to SQL declaration to create datetimetz columns in statements
+     * like CREATE TABLE, for example ' WITH TIME ZONE'.
+     *
+     * Requires a leading space unless empty.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTzTypeSQLSuffix(array $column): string
+    {
+        return '';
+    }
+
+    /**
+     * Obtains DBMS specific suffix to add to SQL declaration to create time columns in statements
+     * like CREATE TABLE, for example ' WITHOUT TIME ZONE'.
+     *
+     * Requires a leading space unless empty.
+     *
+     * @param mixed[] $column
+     */
+    protected function getTimeTypeSQLSuffix(array $column): string
+    {
+        return '';
+    }
+
+    /**
+     * Obtains DBMS specific precision specifier to add to SQL declaration to create datetime columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTypePrecisionSQLSnippet(array $column): string
+    {
+        $scale = empty($column['scale']) || (int) $column['scale'] === Column::DATETIME_SCALE_NOT_SET
```

Comment:
This should use `precision`, not `scale`.

Reply:
Actually, this is weird, I dumped the output of the `$column` array when generating a diff migration, and for some reason `precision` is 10 by default even though the code says it should be 0: https://github.com/doctrine/orm/blob/2.17.x/lib/Doctrine/ORM/Mapping/Column.php#L43

Reply:
I'm also noticing that currently all the Type class' `convertToPHPValue()` methods try to brute-force the conversion instead of just checking the `$column['scale']` (which currently isn't passed in as a parameter to these methods).
They're also missing the opposite `convertToDatabaseValue()` methods, which should be using the same format; `convertToDatabaseValue() -> $value->format($format)`, `convertToPHPValue() -> DateTimeClass:createFromFormat($format, $value)`, where `$format` changes depending on `$column['scale']`.